packages/bun-types/bun.d.ts

```diff
@@ -6127,6 +6139,34 @@ declare module "bun" {
      * The number of bytes written to the socket.
      */
     readonly bytesWritten: number;
+
+    resume(): void;
+
+    pause(): void;
+
+    renegotiate(): void;
+
+    setVerifyMode(requestCert: boolean, rejectUnauthorized: boolean): void;
+
+    getSession(): void;
+
+    setSession(session: string | Buffer | BufferSource): void;
+
+    exportKeyingMaterial(length: number, label: string, context?: string | BufferSource): void;
+
+    upgradeTLS<Data>(options: TLSUpgradeOptions<Data>): [Socket<Data>, Socket<Data>];
```

Comment:
Could we add named keys for the tuple? E.g. `[a: Socket<Data>, b: Socket<Data>]` (replace a/b with the real appropriate names)

Perhaps also some JSDoc for upgradeTLS especially, as it's one of the more requested symbols for us to document